packages/mermaid/src/diagrams/sankey/sankeyRenderer.ts

```diff
@@ -0,0 +1,202 @@
+  sankeyCenter as d3SankeyCenter,
+  sankeyJustify as d3SankeyJustify,
+  SankeyNode as d3SankeyNode,
+} from 'd3-sankey';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import { Uid } from '../../rendering-util/uid.js';
+import { SankeyLinkColor, SankeyNodeAlignment } from '../../config.type.js';
+
+/**
+ * Draws Sankey diagram.
+ *
+ * @param text - The text of the diagram
+ * @param id - The id of the diagram which will be used as a DOM element id¨
+ * @param _version - Mermaid version from package.json
+ * @param diagObj - A standard diagram containing the db and the text and type etc of the diagram
+ */
+export const draw = function (text: string, id: string, _version: string, diagObj: Diagram): void {
+  // Get Sankey config
+  const { securityLevel, sankey: conf } = configApi.getConfig();
+
+  // TODO:
+  // This code repeats for every diagram
+  // Figure out what is happening there, probably it should be separated
+  // The main thing is svg object that is a d3 wrapper for svg operations
+  //
+  let sandboxElement: any;
+  if (securityLevel === 'sandbox') {
+    sandboxElement = d3select('#i' + id);
+  }
+  const root =
+    securityLevel === 'sandbox'
+      ? d3select(sandboxElement.nodes()[0].contentDocument.body)
+      : d3select('body');
+  // @ts-ignore TODO root.select is not callable
+  const svg = securityLevel === 'sandbox' ? root.select(`[id="${id}"]`) : d3select(`[id="${id}"]`);
+
+  // Establish svg dimensions and get width and height
+  //
+  const width = conf?.width || 800;
+  const height = conf?.height || 400;
+  const useMaxWidth = conf?.useMaxWidth || false;
+  const nodeAlignment = conf?.nodeAlignment || SankeyNodeAlignment.justify;
+
+  // FIX: using max width prevents height from being set, is it intended?
+  // to add height directly one can use `svg.attr('height', height)`
+  //
+  // @ts-ignore TODO: svg type vs selection mismatch
+  configureSvgSize(svg, height, width, useMaxWidth);
+
+  // Prepare data for construction based on diagObj.db
+  // This must be a mutable object with `nodes` and `links` properties:
+  //
+  //    {
+  //      "nodes": [ { "id": "Alice" }, { "id": "Bob" }, { "id": "Carol" } ],
+  //      "links": [ { "source": "Alice", "target": "Bob", "value": 23 }, { "source": "Bob", "target": "Carol", "value": 43 } ]
+  //    }
+  //
+  // @ts-ignore TODO: db should be coerced to sankey DB type
+  const graph = diagObj.db.getGraph();
+
+  // Map config options to alignment functions
+  const alignmentsMap: Map<
+    SankeyNodeAlignment,
+    (node: d3SankeyNode<object, object>, n: number) => number
+  > = new Map([
+    [SankeyNodeAlignment.left, d3SankeyLeft],
+    [SankeyNodeAlignment.right, d3SankeyRight],
+    [SankeyNodeAlignment.center, d3SankeyCenter],
+    [SankeyNodeAlignment.justify, d3SankeyJustify],
+  ]);
+  // We need fallback because typescript thinks that `get` can result in undefined
+  const nodeAlign = alignmentsMap.get(nodeAlignment) || d3SankeyJustify;
```

Comment:
Using a Record instead of a map will resolve the TS errors. 

```ts
enum TestEnum {
    hello = 'hello',
    world = 'world'
}

const enumMap: Record<TestEnum, string> = {
    hello: 'awesome',
    world: 'great'
}

const testFn = (key: TestEnum): string => {
    return enumMap[key];
}
```